core/src/test/java/org/apache/iceberg/actions/TestBinPackRewriteFileGroupPlanner.java

```diff
@@ -0,0 +1,468 @@
+
+    Map<String, String> invalidDeleteRatioThresholdOptions =
+        ImmutableMap.of(BinPackRewriteFileGroupPlanner.DELETE_RATIO_THRESHOLD, "127");
+    assertThatThrownBy(() -> planner.init(invalidDeleteRatioThresholdOptions))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("'delete-ratio-threshold' is set to 127.0 but must be <= 1");
+  }
+
+  @Test
+  void testSizeFiltering() {
+    BinPackRewriteFileGroupPlanner planner = new BinPackRewriteFileGroupPlanner(table);
+    FileScanTask tooSmallTask = new MockFileScanTask(100L);
+    FileScanTask optimal = new MockFileScanTask(450);
+    FileScanTask tooBigTask = new MockFileScanTask(1000L);
+    List<FileScanTask> tasks = ImmutableList.of(tooSmallTask, optimal, tooBigTask);
+
+    Map<String, String> options =
+        ImmutableMap.of(
+            BinPackRewriteFileGroupPlanner.MIN_FILE_SIZE_BYTES, "250",
+            BinPackRewriteFileGroupPlanner.TARGET_FILE_SIZE_BYTES, "500",
+            BinPackRewriteFileGroupPlanner.MAX_FILE_SIZE_BYTES, "750");
+    planner.init(options);
+
+    Iterable<List<FileScanTask>> groups = planner.planFileGroups(tasks);
+    assertThat(groups).as("Must have 1 group").hasSize(1);
+    List<FileScanTask> group = Iterables.getOnlyElement(groups);
+    assertThat(group).as("Must rewrite 2 files").hasSize(2);
+    assertThat(group).containsExactlyInAnyOrder(tooSmallTask, tooBigTask);
+  }
+
+  @Test
+  void testGroupWithEnoughFiles() {
+    BinPackRewriteFileGroupPlanner planner = new BinPackRewriteFileGroupPlanner(table);
+    List<FileScanTask> tasks =
+        ImmutableList.of(
+            new MockFileScanTask(100L),
+            new MockFileScanTask(100L),
+            new MockFileScanTask(100L),
+            new MockFileScanTask(100L));
+
+    Map<String, String> options =
+        ImmutableMap.of(
+            BinPackRewriteFileGroupPlanner.MIN_INPUT_FILES, "3",
+            BinPackRewriteFileGroupPlanner.MIN_FILE_SIZE_BYTES, "150",
+            BinPackRewriteFileGroupPlanner.TARGET_FILE_SIZE_BYTES, "1000",
+            BinPackRewriteFileGroupPlanner.MAX_FILE_SIZE_BYTES, "5000");
+    planner.init(options);
+
+    Iterable<List<FileScanTask>> groups = planner.planFileGroups(tasks);
+    assertThat(groups).as("Must have 1 group").hasSize(1);
+    List<FileScanTask> group = Iterables.getOnlyElement(groups);
+    assertThat(group).as("Must rewrite 4 files").hasSize(4);
+  }
+
+  @Test
+  void testGroupWithEnoughData() {
+    BinPackRewriteFileGroupPlanner planner = new BinPackRewriteFileGroupPlanner(table);
+    List<FileScanTask> tasks =
+        ImmutableList.of(
+            new MockFileScanTask(100L), new MockFileScanTask(100L), new MockFileScanTask(100L));
+
+    Map<String, String> options =
+        ImmutableMap.of(
+            BinPackRewriteFileGroupPlanner.MIN_FILE_SIZE_BYTES, "200",
+            BinPackRewriteFileGroupPlanner.TARGET_FILE_SIZE_BYTES, "250",
```

Comment:
should this be larger than 300 so the the MIN_FILE_SIZE_BYTES is the trigger condition?

Reply:
The number of planned groups depends on the `MAX_FILE_SIZE_BYTES`. The number of the files we would like to rewrite depends on the `MIN_FILE_SIZE_BYTES` and `TARGET_FILE_SIZE_BYTES`.

Since all of the files are 100 bytes long in the test, all of them will be target for rewrite. This is true for both `200` or `300`. The only difference is the resulting `expectedOutputFiles` which should be 2 with `200` and 1 with `300`.